metrics/cli/test/deploy/deployer_test.dart

```diff
@@ -621,10 +742,167 @@ void main() {
+            .thenAnswer((_) => Future.value(sentryRelease));
+
+        await deployer.deploy();
+
+        verify(sentryService.login()).called(once);
+      },
+    );
+
+    test(
+      ".deploy() does not log in to the Sentry if the user does not agree with the Sentry setup",
+      () async {
+        whenDirectoryExist().thenReturn(true);
+        whenPromptToSetupSentry().thenReturn(false);
+
+        await deployer.deploy();
+
+        verifyNever(sentryService.login());
+      },
+    );
+
+    test(
+      ".deploy() deletes the temporary directory if Sentry service throws during the login process",
+      () async {
+        whenDirectoryExist().thenReturn(true);
+        whenPromptToSetupSentry().thenReturn(true);
+        when(sentryService.login()).thenAnswer((_) => Future.error(stateError));
+
+        await expectLater(deployer.deploy(), throwsStateError);
+
+        verify(directory.deleteSync(recursive: true)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() logs in to the Sentry before creating the Sentry release",
+      () async {
+        whenDirectoryExist().thenReturn(true);
+        whenPromptToSetupSentry().thenReturn(true);
+        whenCreateSentryRelease()
+            .thenAnswer((_) => Future.value(sentryRelease));
+
+        await deployer.deploy();
+
+        verifyInOrder([
+          sentryService.login(),
+          sentryService.createRelease(any),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() creates the Sentry release",
+      () async {
+        whenDirectoryExist().thenReturn(true);
+        whenPromptToSetupSentry().thenReturn(true);
+        whenCreateSentryRelease()
+            .thenAnswer((_) => Future.value(sentryRelease));
+
+        await deployer.deploy();
+
+        verify(sentryService.createRelease(any)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() deletes the temporary directory if Sentry service throws during the release creation",
+      () async {
+        whenGetDirectory().thenReturn(directory);
+        whenDirectoryExist().thenReturn(true);
+        whenPromptToSetupSentry().thenReturn(true);
+        whenCreateSentryRelease().thenAnswer((_) => Future.error(stateError));
+
+        await expectLater(deployer.deploy(), throwsStateError);
+
+        verify(directory.deleteSync(recursive: true)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() creates the Sentry release before requesting the Sentry project DSN",
+      () async {
+        whenDirectoryExist().thenReturn(true);
+        whenPromptToSetupSentry().thenReturn(true);
+        whenCreateSentryRelease()
+            .thenAnswer((_) => Future.value(sentryRelease));
+
+        await deployer.deploy();
+
+        verifyInOrder([
+          sentryService.createRelease(any),
+          sentryService.getProjectDsn(any),
+        ]);
+      },
+    );
+
+    test(
+      ".deploy() requests the Sentry DSN using the Sentry project within the created Sentry release",
```

Comment:
```suggestion
      ".deploy() requests the Sentry DSN of the created project",
```
What do you think? Also, should we test that this method will remove the temp directory if the Prompter throws an error?